hazelcast/src/main/java/com/hazelcast/sql/impl/expression/ColumnExpression.java

```diff
@@ -76,7 +81,12 @@ public T eval(Row row, ExpressionEvalContext context) {
 
     private Object unwrapLazyValue(LazyTarget lazyValue, ExpressionEvalContext context) {
         assert type.equals(QueryDataType.OBJECT);
-
+        Data serialized = lazyValue.getSerialized();
+        boolean dataIsCompactOrPortable = serialized != null
+                && (serialized.isCompact() || serialized.isPortable());
```

Comment:
May be we should Serializable and Externalizable as well to this list. 
    public static final int JAVA_DEFAULT_TYPE_SERIALIZABLE = -100;
    public static final int JAVA_DEFAULT_TYPE_EXTERNALIZABLE = -101;